docs/rfcs/1011-non-blocking-invite-transport.md

```diff
@@ -0,0 +1,1226 @@
+            {
+                "status": "ok"
+            },
+            {
+                "status": "invitation_not_found"
+            },
+            {
+                "status": "invitation_already_completed"
+            },
+            {
+                "status": "invitation_cancelled"
+            }
+        ]
+    }
+]
+```
+
+
+### Removed routes
+
+All the `invite_<N>_claimer_<step name>` and `invite_<N>_greeter_<step name>` are removed and replaced by `invite_claimer_step` and `invite_greeter_step`.
+
+This includes:
+- `invite_1_greeter_wait_peer`
+- `invite_2a_greeter_get_hashed_nonce`
+- `invite_2b_greeter_send_nonce`
+- `invite_3a_greeter_wait_peer_trust`
+- `invite_3b_greeter_signify_trust`
+- `invite_4_greeter_communicate`
+- `invite_1_claimer_wait_peer`
+- `invite_2a_claimer_send_hashed_nonce`
+- `invite_2b_claimer_send_nonce`
+- `invite_3a_claimer_signify_trust`
+- `invite_3b_claimer_wait_peer_trust`
+- `invite_4_claimer_communicate`
+
+
+### Protocol update
+
+Here is an overview of the different kinds of data that can be contained in a claimer or greeter step:
+- public key: `PublicKey`
+- nonce: `Bytes`
+- hashed nonce: `HashDigest`
+- acknowledgement: `None`
+- payload: `Bytes`
+
+The news steps are defined as such:
+
+| Index | Greeter step       | Greeter data   | Claimer step        | Claimer data   |
+|-------|--------------------|----------------|---------------------|----------------|
+| 0     | `wait_peer`        | public key     | `wait_peer`         | public key     |
+| 1     | `get_hashed_nonce` |                | `send_hashed_nonce` | hashed nonce   |
+| 2     | `send_nonce`       | nonce          | `get_nonce`         |                |
+| 3     | `get_nonce`        |                | `send_nonce`        | nonce          |
+| 4     | `wait_peer_trust`  |                | `signify_trust`     | acknowledgment |
+| 5     | `signify_trust`    | acknowledgment | `wait_peer_trust`   |                |
+| 6     | `get_payload`      |                | `send_payload`      | payload        |
+| 7     | `send_payload`     | payload        | `get_payload`       |                |
+| 8     | `wait_peer_ack`    |                | `acknowledge`       | acknowledgment |
+
+Those steps work very similarly to Parsec v2. There are a couple of differences though:
+- Steps are not longer counted as `1`, `2a`, `2b`, `3a`, `3b`, and `4` like in version 2
+- Three different exchanges were performed in steps `2a` and `2b`. Now those exchanges are expanded in steps `1`, `2` and `3`.
+- Step `4` in version 2 could be used for several exchanges, until the greeter marked an exchange with `last=True`. Instead, those exchanges are expanded in steps `6` and `7`
+- In version 2, the last step was used to complete the invite. In version 3 this is now done with a dedicated `invite_complete` command, so we add a last step (index `8`) for the claimer to acknowledge that it's been able to successfully process the greeter payload.
```

Comment:
We don't need a dedicated step in the table above, because by using the `invite_complete` command we mutate the step of the invitation to `complete`, Am I right?

Reply:
Yes because a completed attempt is different from a completed invitation (especially for the shamir):

See in the `invite_complete` section:

> In the previous implementation, this was done automatically when the last exchange was performed. However, this is confusing in some cases:
> - Maybe the last exchange worked but the reply didn't reach one of the peers. In this case, the peer cannot retry the exchange since the invite is now completed.
> - In a shared recovery (shamir) invitation, the invitation must not complete after a successful attempt since there might be more exchanges to complete with other recipients.